contrib/epee/include/storages/portable_storage_template_helper.h

```diff
@@ -84,7 +85,7 @@ namespace epee
     }
     //-----------------------------------------------------------------------------------------------------------
     template<class t_struct>
-    bool load_t_from_binary(t_struct& out, const epee::span<const uint8_t> binary_buff)
+    bool load_t_from_binary(t_struct& out, const epee::span<const uint8_t> & binary_buff)
```

Comment:
Drop the `&`, this was not a mistake or oversight. If this was done elsewhere in the patch too, which is unrelated to the PR title, please revert them. There are numerous internet discussions about using `string_view`/`string_ref`/`span` objects by value instead of reference. The type is has 1 pointer and 1 `size_t` value (small), and is trivial to copy and destruct.

A quick summary of the discussions - this type by value requires setting two registers on all modern ABIs, and even on legacy x86 passing the value is fairly simple. In the majority of cases, this object is copying values from a `std::string` or `std::vector` so a `&` forces the compiler to push space onto the stack, copy a pointer+length, then set a register to that stack location. Then the backend optimizer is forced to consider aliasing with other `span` objects of this type (which should rarely be an issue, but why bother). In contrast, by value requires setting two registers and the backend knows the `span<>` type itself cannot be aliased.

Reply:
It's not the size of the construct, that matters to me in this regard, but the impossibility of forward declaring a class or a template class, which is being passed by a value. I can prove to which extent a forward declaration of a moderately heavy template improves, but whenever discussing such topic with people whose priority is runtime optimization, they rarely bring up any measurements.

In this case, I reverted the change, as I was forced to include the span.h anyway, because it's constructor is being used elsewhere. Template segmentation will be the answer to this problem, but I don't want to make too large changes in one go.

Reply:
After gathering a 2nd opinion, I switched the span reference back to by-value.